kong/runloop/plugins_iterator.lua

```diff
@@ -321,14 +321,28 @@ function PluginsIterator.new(version)
     end
 
     if should_process_plugin(plugin) then
-      map[plugin.name] = true
+      local name = plugin.name
+
+      map[name] = true
 
       local combo_key = (plugin.route    and 1 or 0)
                       + (plugin.service  and 2 or 0)
                       + (plugin.consumer and 4 or 0)
 
-      combos[plugin.name] = combos[plugin.name] or {}
-      combos[plugin.name][combo_key] = true
+      combos[name]          = combos[name]          or {}
+      combos[name].both     = combos[name].both     or {}
+      combos[name].routes   = combos[name].routes   or {}
+      combos[name].services = combos[name].services or {}
+
```

Comment:
Is the reason behind not including `consumers` in this new heuristic because of the high-cardinality of the Consumer ids attribute? Because as the PR stands, the same issue with DB connectivity would occur when the request is authenticated, but the plugin currently being evaluated by the runloop doesn't apply to the authenticated Consumer.

Reply:
@thibaultcha, yes, the high cardinality was the reason. In one proposal at one point I proposed that we just load all the plugin configurations in memory, but that was not accepted at that time because of the high-cardinality. Of course we were then speaking about caching the whole plugin configurations, and not just ids, as what this PR does. So with caching just ids the memory pressure is not that big, but well, high-cardinality of consumers may still be seen as a problem. Perhaps we can leave that decision later, and first release without, and rethink about caching the plugin consumer.ids later (just thinking that perhaps moving one step at time is safer, and we may get feedback on it). I can create a backlog item on that. One obviously good thing about caching the ids is that it makes negative caching not needed, so if we add consumer ids to this cache too, we will not need negative caching with plugins at all.  If you feel strongly about it, let me know, the caching of consumer ids is rather easy to add.